Java/libraries/recognizers-text-number-with-unit/src/main/java/com/microsoft/recognizers/text/numberwithunit/parsers/BaseCurrencyParser.java

```diff
@@ -120,37 +125,22 @@ private ParseResult mergeCompoundUnit(ExtractResult compoundResult) {
                 } else {
                     // If the fraction unit doesn't match the main unit, finish process this group.
                     if (result != null) {
-                        if (mainUnitIsoCode == null || mainUnitIsoCode.isEmpty() ||
-                            mainUnitIsoCode.startsWith(Constants.FAKE_ISO_CODE_PREFIX)) {
-                            result.setValue(new UnitValue(String.valueOf(numberValue), mainUnitValue));
-                        } else {
-                            result.setValue(new CurrencyUnitValue(String.valueOf(numberValue), mainUnitValue, mainUnitIsoCode));
-                        }
-
-                        results.add(result);
-                        result = null;
+                        addToResults(result,mainUnitIsoCode,numberValue,mainUnitValue,results);
                     }
+                    result = null;
 
                     count = 0;
                     idx -= 1;
+                    numberValue = -1;
                     continue;
                 }
             }
 
             count++;
         }
-
```

Comment:
Don't remove this blank line. Always keep a blank line between unrelated if brackets.